### PR TITLE
Closes #1725 - `sqrt` and `power` functions on pdarrays

### LINKED ISSUE #1725: Add sqrt and power functions to operate on pdarray
Square root and power operations are useful; this enhancement would add them as functions to the `arkouda` module alongside `exp` etc.

`arkouda.sqrt` would be a simple element-wise operation (like `exp`). 
`arkouda.power` would have a similar interface to [`numpy.power`](https://numpy.org/doc/stable/reference/generated/numpy.power.html), accepting as the second argument either a scalar value to which all elements of the first array are raised, or an array of powers to which corresponding elements of the first array are raised.

### PR DESCRIPTION
**Pull request by Ethan-DeBandi99**
Closes #1725 

Added `ak.power` and `ak.sqrt` functions. These functions are just aliases for `**` and `**0.5`, respectively. 
Adds testing to ensure arkouda functionality is in alignment with `Numpy`.

### CODE REVIEW

File: tests/operator_tests.py [hunk 1]
```diff
@@ -453,6 +453,33 @@ def equality_helper(a, n):
             equality_helper(ak_edge_cases // a_vect, np_edge_cases // n_vect)
             equality_helper(a_vect // ak_edge_cases, n_vect // np_edge_cases)
 
+    def test_pda_sqrt(self):
+        n = np.array([4, 16, -1, 0])
```
Comment (pierce314159) [inline tests/operator_tests.py:457]: we should add `np.inf` here to make sure it remains infinity. Perhaps `-np.inf`, `-0.0` too to ensure we're covering our edge cases well

File: tests/operator_tests.py [hunk 2]
```diff
@@ -453,6 +453,33 @@ def equality_helper(a, n):
...
+        # id nans
+        nan = ak.isnan(s)
+        npnan = np.isnan(ex)
+        self.assertListEqual(nan.to_list(), npnan.tolist())
+
+        # check results are equal remove nans
+        self.assertListEqual(s[~nan].to_list(), ex[~npnan].tolist())
+
+    def test_pda_power(self):
+        n = np.array([10, 5, 2])
```
Comment (pierce314159) [inline tests/operator_tests.py:472]: yeah i think testing some of the weirder edge cases would be good, maybe `inf`, `-inf`, `0.0`, and `-0.0`

**Review by pierce314159 — COMMENTED** [review]
A couple things to look into

Also numpy has a `where` argument for both of these functions, but there's some design questions i have about it's implementation, so i think adding it is outside the scope of this PR. I've captured this in issue #1727

File: tests/operator_tests.py (hunk 2 above)
Comment (Ethan-DeBandi99) [reply tests/operator_tests.py:472]: Added

File: tests/operator_tests.py (hunk 1 above)
Comment (pierce314159) [reply tests/operator_tests.py:457]: Note chapel doesn't handle `(-INFINITY)**0.5` and `(-0.0)**0.5` correctly. These cases are correctly handled by the chapel `sqrt` function. I put up https://github.com/chapel-lang/chapel/issues/20542 to track this.

We should be able to get around this on the client in the meantime with a bit of hacky code... something like
```python
ret_array = # normal stuff
ret_array[pda == -np.inf] = np.nan
ret_array[pda== -0.0] = -0.0
```

Comment (pierce314159) [reply tests/operator_tests.py:457]: digging in a bit deeper, it seems numpy also has variation here...
```python
>>> n = np.array([-0.0, -7, -np.inf])
>>> n**0.5
array([-0., nan, nan])

>>> np.power(n, 0.5)
array([ 0., nan, inf])

>>> np.sqrt(n)
array([-0., nan, nan])

>>> n**(1/3)
array([ 0., nan, inf])

>>> np.power(n, 1/3)
array([ 0., nan, inf])

>>> n**(1/4)
array([ 0., nan, inf])

>>> np.power(n, 1/4)
array([ 0., nan, inf])
```

Comment (Ethan-DeBandi99) [reply tests/operator_tests.py:457]: I will be adding a ticket to track this once I push the other requested changes.

Comment (Ethan-DeBandi99) [reply tests/operator_tests.py]: Added issue #1730

**Review by pierce314159 — APPROVED** [review]

**Review by joshmarshall1 — APPROVED** [review]
